Proj kon/task1

HubertKarw opened the pull request.

HubertKarw committed:
first commit

HubertKarw committed:
proj-kon/task1/changelog
added liquibase changelog to create tables and constraints for the product and customization
additionally added PRODUCT_CUSTOMIZATION table to connect customizations with the product

HubertKarw committed:
proj-kon/task1/models,repos, liquibase changes and Dockerfile, docker-compose.yaml creation for testing

HubertKarw committed:
proj-kon/task1/controllers, services, error massage model, exceptions and exception handler added
+ formatting

HubertKarw committed:
proj-kon/task1/mappers

HubertKarw committed:
proj-kon/task1/controllers and creation commands for entities

HubertKarw committed:
proj-kon/task1/controllers services implementation

konradgab commented on src/main/java/com/hubertkarw/product/controller/CustomizationController.java:
```diff
@@ -0,0 +1,26 @@
+
+    @PostMapping
+    CustomizationDTO addCustomization(){
+        return null;
```
todo

konradgab commented on src/main/java/com/hubertkarw/product/controller/ProductController.java:
```diff
@@ -0,0 +1,38 @@
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/products")
+public class ProductController {
```
Nie zapominaj o SWAGGER

konradgab commented on src/main/java/com/hubertkarw/product/exception/ProductAppExceptionHandler.java:
```diff
@@ -0,0 +1,27 @@
+import java.time.LocalDateTime;
+
+@ControllerAdvice
+
```
pusta linia

konradgab commented on src/main/java/com/hubertkarw/product/service/CustomizationService.java:
```diff
@@ -0,0 +1,27 @@
+    private final CustomizationMapper mapper;
+
+    public List<CustomizationDTO> getCustomizations(String type) {
+        if(type==null){
```
to moznaby ladniej switchem np

konradgab commented on src/main/java/com/hubertkarw/product/service/ProductService.java:
```diff
@@ -0,0 +1,49 @@
+    private final ProductMapper mapper;
+
+    public List<ProductDTO> getProducts(String type) {
+        if (type == null) {
```
return type == null ? findAll() : findByType(type);

konradgab commented on src/main/java/com/hubertkarw/product/service/ProductService.java:
```diff
@@ -0,0 +1,49 @@
+        return mapper.toDTO(repository.save(mapper.toEntity(productDTO)));
+    }
+
+    public ProductDTO updateProduct(Long id, ProductCreateDTO productDTO) {
```
@Transactional

konradgab commented on src/main/java/com/hubertkarw/product/service/ProductService.java:
```diff
@@ -0,0 +1,49 @@
+                .toList();
+    }
+
+    public ProductDTO addProduct(ProductCreateDTO productDTO) {
```
@Transactional

konradgab commented on src/main/java/com/hubertkarw/product/model/Customization.java:
```diff
@@ -0,0 +1,23 @@
+    private long id;
+    private String name;
+    private String type;
+    private BigDecimal price;
```
private Long productId // powiazanie konfiguracji z produktym którym jest jesli jest sprzedawana oddzielnie

konradgab commented on src/main/resources/liquibase/changelog/000001_product.yaml:
```diff
@@ -0,0 +1,89 @@
+            tableName: PRODUCT_CUSTOMIZATION
+            columns:
+              - column:
+                  name: PRODUCT_ID
```
zweryfikuj czy to tworzy 2 PK czy jeden kompozytowy

konradgab commented on src/main/resources/liquibase/changelog/000001_product.yaml:
```diff
@@ -0,0 +1,89 @@
+                  constraints:
+                    nullable: false
+              - column:
+                  name: PRODUCT_ID
```
add FK

HubertKarw committed:
proj-kon/task1/customization and product functionality review fixes, new Exception classes logging and swagger
created functionality for product and customization
review comments fixed
added ProductNotFoundException and CustomizationNotFountException
added logging in controllers and error handlers
added swagger descriptions

HubertKarw committed:
proj-kon/task1/transactional equals and hashCode
Added Transactional where needed
Created equals and hashCode for product and customization

HubertKarw committed:
proj-kon/task1/tests
testing classes initialized and first controller test for customization controller

HubertKarw committed:
proj-kon/task1/tests
implementation og CustomizationControllerTest, CustomizationServiceTest, ProductControllerTest

HubertKarw committed:
proj-kon/task1/tests
implementation of ProductServiceTest
small changes in CustomizationServiceTest

HubertKarw committed:
proj-kon/task1/client
added client and its functionality

konradgab commented on pom.xml:
```diff
@@ -65,6 +84,12 @@
+		<dependency>
+			<groupId>org.springdoc</groupId>
+			<artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
+			<version>2.8.5</version>
```
mozna wyniesc wersje do zmiennej

konradgab commented on src/main/java/com/hubertkarw/product/controller/ProductController.java:
```diff
@@ -0,0 +1,170 @@
+package com.hubertkarw.product.controller;
+
+
```
pusta linia

konradgab commented on src/main/java/com/hubertkarw/product/model/Customization.java:
```diff
@@ -0,0 +1,48 @@
+    private String name;
+    private String type;
+    private BigDecimal price;
+    private Long productId;
```
zrób tutaj normlanie relacje

konradgab commented on src/main/java/com/hubertkarw/product/service/CustomizationService.java:
```diff
@@ -0,0 +1,50 @@
+    private final CustomizationMapper mapper;
+
+    public List<CustomizationDTO> getCustomizations(String type) {
+        return type==null?mapper.toDTOList(repository.findAll()):mapper.toDTOList(repository.findByType(type));
```
ctrl alt l

konradgab commented on src/main/java/com/hubertkarw/product/service/ProductService.java:
```diff
@@ -0,0 +1,131 @@
+
+import static java.util.stream.Collectors.toList;
+
+
```
nadmiarowa pusta linia

konradgab commented on src/main/java/com/hubertkarw/product/service/ProductService.java:
```diff
@@ -0,0 +1,131 @@
+                .map(Customization::getId)
+                .toList();
+
+        for (Long customizationId : customizationIds) {
```
to jest metoda publiczna w serwisie stworzmy metode ktorej nazwa powie nam co to robi

konradgab commented on src/main/java/com/hubertkarw/product/service/ProductService.java:
```diff
@@ -0,0 +1,131 @@
+
+    //TODO TESTING
+    @Transactional
+    public CartItemRequest addProductToCart(Long id, Long cartId, List<Long> customizationIds) {
```
wydaje mi sie to niepotrzebne
